Assets/HoloToolkit/Utilities/Scripts/SceneLauncher.cs

```diff
@@ -0,0 +1,133 @@
+                        GameObject.Destroy(returnToSceneLauncher);
+                    };
+                    returnToSceneLauncher.KeywordsAndResponses[0].Response.AddListener(keywordAction);
+                }
+            }
+
+            // Determine the size of the buttons. Instantiate one of them so that we can check its bounds.
+            Interactive sceneButtonForSize = Instantiate(SceneButtonPrefab);
+            Collider sceneButtonForSizeCollider = sceneButtonForSize.GetComponent<Collider>();
+            if (sceneButtonForSizeCollider != null)
+            {
+                SetSceneButtonWidthScale(sceneButtonForSize);
+                sceneButtonSize = sceneButtonForSizeCollider.bounds.size;
+            }
+            Destroy(sceneButtonForSize.gameObject);
+
+            // Create an empty game object to serve as a parent for all the buttons we're about to create.
+            GameObject buttonParent = new GameObject("Buttons");
+
+            List<string> sceneNames = SceneList.Instance.GetSceneNames();
+            for (int iScene = 0; iScene < sceneNames.Count; ++iScene)
+            {
+                CreateSceneButton(buttonParent, iScene, sceneNames);
+            }
+        }
+
+        private void CreateSceneButton(GameObject buttonParent, int iScene, List<string> sceneNames)
+        {
+            string sceneName = sceneNames[iScene];
+            Scene scene = SceneManager.GetSceneByBuildIndex(iScene);
+            Debug.Assert(SceneManager.GetSceneByName(sceneName) == scene);
+
+            Interactive sceneButton = Instantiate<Interactive>(SceneButtonPrefab, GetButtonPosition(iScene, sceneNames.Count), Quaternion.identity, buttonParent.transform);
+            sceneButton.name = sceneName;
+            SetSceneButtonWidthScale(sceneButton);
+            sceneButton.IsEnabled = scene != SceneManager.GetActiveScene(); // Disable button to launch our own scene.
+            int buildIndex = iScene;
+            UnityAction buttonAction = delegate
+            {
+                Debug.LogFormat("SceneLauncher: Loading scene {0}: {1}", buildIndex, SceneList.Instance.GetSceneNames()[buildIndex]);
+                SceneManager.LoadScene(buildIndex, LoadSceneMode.Single);
+            };
+            sceneButton.OnSelectEvents.AddListener(buttonAction);
+            LabelTheme labelTheme = sceneButton.GetComponent<LabelTheme>();
+            if (labelTheme != null)
+            {
+                labelTheme.Default = sceneName;
+            }
+        }
+
+        private void SetSceneButtonWidthScale(Interactive sceneButton)
+        {
+            // Scale the button horizontally by SceneButtonWidthScale to make more space for text.
+            sceneButton.transform.localScale = Vector3.Scale(sceneButton.transform.localScale, new Vector3(SceneButtonWidthScale, 1.0f, 1.0f));
+            foreach (TextMesh textMesh in sceneButton.GetComponentsInChildren<TextMesh>())
```

Comment:
nit:
```
Vector 3 buttonScale = new Vector3(1.0f / SceneButtonWidthScale, 1.0f, 1.0f);
TextMesh[] childTextMeshes = sceneButton.GetComponentsInChildren<TextMesh>();
for (int i=0; i < childTextMeshes.Length; i++)
    {
    // Reverse the scale applied to the button so that the text isn't stretched by the scale.
   childTextMeshes[i].transform.localScale = Vector3.Scale(childTextMeshes[i].transform.localScale, buttonScale));
    }
```